bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandThermostatHandler.java

```diff
@@ -58,28 +58,34 @@ void touchWandUnitHandleCommand(Command command) {
         if (touchWandBridgeHandler != null) {
             if (command instanceof OnOffType) {
                 touchWandBridgeHandler.touchWandClient.cmdThermostatOnOff(unitId, (OnOffType) command);
-            } else {
-                String sCommand = command.toString();
-                switch (sCommand) {
-                    case "cool":
-                    case "heat":
-                    case "fan":
-                    case "auto":
-                    case "dry":
-                        touchWandBridgeHandler.touchWandClient.cmdThermostatMode(unitId, sCommand);
-                        break;
-                    case "low":
-                    case "medium":
-                    case "high":
-                        touchWandBridgeHandler.touchWandClient.cmdThermostatFanLevel(unitId, sCommand);
-                        break;
-                    case "fanAuto":
-                        touchWandBridgeHandler.touchWandClient.cmdThermostatFanLevel(unitId, "auto");
-                        break;
-                    default:
-                        touchWandBridgeHandler.touchWandClient.cmdThermostatTargetTemperature(unitId, sCommand);
-                        break;
-                }
+                return;
+            }
+            if (command instanceof QuantityType) {
+                QuantityType<?> value = (QuantityType<?>) command;
+                String targetTemperature = String.valueOf(value.intValue());
```

Comment:
It seems that you are simply stripping the unit. Which unit type is expected here, °C or °F? Perhaps you need to convert to target unit before applying the value, something like:

```
final QuantityType<?> value = ((QuantityType<?>) command).toUnit(SIUnits.CELSIUS);
```

Reply:
good idea , I converted to C and then apply the value